### PR TITLE
Refactor/create fetch file

### LINKED ISSUE #14: Remove Ingredients from Pantries
**What is the proposed feature? Please describe user interaction.**
Remove the ingredients used for a given meal from my pantry, once that meal has been cooked (only applicable if users have a list of mealsToCook; can be considered a stretch goal)

**Additional context**
Not sure what exactly this would look like, how would the user mark a meal as cooked?

### LINKED ISSUE #46: Post Functionality
**Description of what needs refactoring**
Finalize post functionality for updating user pantry

**Additional Context** 
Should be able to make the post function more dynamic so it can be used to add and remove items from the pantry 
Add a button to the recipe modal to add ingredients to pantry
Pantry should be rebuilt after items are added or removed

### LINKED ISSUE #47: FetchAPI file
**Description of what needs refactoring**
The fetch functions should live in their own file to be referenced within scripts

**Additional Context** 
Potential for using promise.all()

**Screenshots**
Screen Shot 2021-01-10 at 6 14 11 PM

### PR DESCRIPTION
**Pull request by alia-peterson**
## *Purpose:*
*What does this merge do? Check all that apply.*
- [ ] Adds a new feature
- [x] Improves an existing feature
- [x] Fixes a bug
- [ ] Cleans up (logs, tests, etc.)

## *Description: What does this PR do?*
- Moved fetch functionality to own file
- Updated post function to be dynamic so that it can work for adding ingredients or removing
- Added shopping list button

## *Instructions for reviewer:*
*Where should the reviewer start?*
scripts.js and fetchApi.js
*How should this be manually tested?*
Check that pantry values are updating when the shop and cook buttons are selected
*What are the relevant tickets?*
Closes #14, closes #46, closes #47

## *Issues:*
No known bugs
Message displaying whether the user has enough ingredients isn't displaying, will need to figure out why in the next refactor ticket

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have used tests to prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my change

### CODE REVIEW

File: src/fetchApi.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+const fetchApi = {
```
Comment (tashiad) [inline src/fetchApi.js:1]: This was totally the move. So clean!

File: src/scripts.js [hunk 2]
```diff
@@ -2,6 +2,7 @@
 import recipeData from  './data/recipe-data';
```
Comment (tashiad) [inline src/scripts.js:2]: Can probably get rid of this by now yeah?

File: src/scripts.js [hunk 3]
```diff
@@ -44,92 +38,102 @@ myPantryButton.addEventListener("click", togglePantryDisplay)
...
-        const newRecipe = new Recipe(recipe)
-
-        allRecipes.push(newRecipe)
-      })
-    })
+const fetchedUserData = fetchApi.getUserData()
+const fetchedRecipeData = fetchApi.getRecipeData()
+const fetchedIngredientData = fetchApi.getIngredientsData()
+
+Promise.all([fetchedUserData, fetchedRecipeData, fetchedIngredientData])
```
Comment (tashiad) [inline src/scripts.js:53]: Promise.all FTW! Definitely want to walk through all of this together later today.

**Review by richardltyler — APPROVED** [review]
God this is so much cleaner. I appreciate you taking care of some of the little stuff like double/single quotes along the way too

- [ ] Follows Turing Style Guidelines
- [ ] Changes generate no new warnings or errors
- [x] No spelling errors
- [x] Approved changes with team
- [x] All teammates understand changes

File: src/css/_recipe-modal.scss [hunk 4]
```diff
@@ -16,37 +16,81 @@
...
-.modal--bottom {
-  align-items: center;
-  display: flex;
-  flex-direction: column;
-  min-height: 95px;
-  justify-content: flex-end;
-
-  p {
+.modal {
+  &--bottom {
```
Comment (tashiad) [inline src/css/_recipe-modal.scss:20]: this is v cool!!! 🤯

File: src/css/_recipe-modal.scss [hunk 5]
```diff
@@ -16,37 +16,81 @@
...
+      font-weight: bold;
+    }
+  }
+
+  &--title {
+    background-repeat: no-repeat;
+    background-size: cover;
+    border-radius: 7px 7px 0px 0px;
+    color: white;
+    font-size: 30px;
```
Comment (tashiad) [inline src/css/_recipe-modal.scss:71]: is px frowned upon? vs using rems or ems or %.

File: src/domUpdates.js [hunk 6]
```diff
@@ -153,6 +151,20 @@ let domUpdates = {
     return instructionsList
   },
 
+  // can probably rework these to be dynamic since they're so similar?
```
Comment (tashiad) [inline src/domUpdates.js:154]: good call!

**Review by tashiad — APPROVED** [review]

**Pull request merged**